src/systemdhook.c

```diff
@@ -69,6 +71,86 @@ static int makepath(char *dir, mode_t mode)
     return mkdir(dir, mode);
 }
 
+static int mount_cgroup_path(const char *rootfs, const char* cgroup_path)
+{
+	char cont_cgroup_dir[PATH_MAX];
+	snprintf(cont_cgroup_dir, PATH_MAX, "%s/%s", rootfs, cgroup_path);
+
+	int ret = 0;
+	if (makepath(cont_cgroup_dir, 0755) == -1) {
+		if (errno != EEXIST) {
+			pr_perror("Failed to mkdir container cgroup dir %s", cgroup_path);
+			return -1;
+		}
+	}
+
+	pr_pinfo("Mounting %s at %s\n", cgroup_path, cont_cgroup_dir);
+	if (mount(cgroup_path, cont_cgroup_dir, "bind", MS_BIND, NULL) == -1) {
+		pr_perror("Failed to mount %s at %s", cgroup_path, cont_cgroup_dir);
+		return -1;
+	}
```

Comment:
Yeah I haven't touched this PR in a bit. I can make the changes or we can use your new PR. 

Sent from my iPhone

> On Feb 16, 2016, at 9:22 PM, Daniel J Walsh <notifications@github.com> wrote:
> 
> In src/systemdhook.c:
> 
> > +	char cont_cgroup_dir[PATH_MAX];
> > +	snprintf(cont_cgroup_dir, PATH_MAX, "%s/%s", rootfs, cgroup_path);
> > +
> > +	int ret = 0;
> > +	if (makepath(cont_cgroup_dir, 0755) == -1) {
> > +		if (errno != EEXIST) {
> > +			pr_perror("Failed to mkdir container cgroup dir %s", cgroup_path);
> > +			return -1;
> > +		}
> > +	}
> > +
> > +	pr_pinfo("Mounting %s at %s\n", cgroup_path, cont_cgroup_dir);
> > +	if (mount(cgroup_path, cont_cgroup_dir, "bind", MS_BIND, NULL) == -1) {
> > +		pr_perror("Failed to mount %s at %s", cgroup_path, cont_cgroup_dir);
> > +		return -1;
> > +	}
> These all need to be mounted read/only other then /sys/fs/cgroup/systemd.
> 
> —
> Reply to this email directly or view it on GitHub.
>